### PR TITLE
ENH: Allow to construct events from annotations

### PR DESCRIPTION
**Pull request by dengemann**
In preparation for improved NEO integration this PR proposes a new method for BaseRaw that allows to create events and event_id from existing annotations. As we currently do not yet have testing data and readers which parse annotations in the relevant way, I do not yet propose examples. Instead I wrote a series of tests based on Annotations that I created from events.

@agramfort @massich please feel free to push / continue.

### CODE REVIEW

File: mne/io/tests/test_raw.py [hunk 1]
```diff
@@ -137,3 +139,61 @@ def test_time_index():
...
+    event_id = {
+        'Auditory/Left': 1,
+        'Auditory/Right': 2,
+        'Visual/Left': 3,
+        'Visual/Right': 4,
+        'Visual/Smiley': 32,
+        'Motor/Button': 5
+    }
+    event_map = {v: k for k, v in event_id.items()}
+    raw.annotations = Annotations(
```
Comment (massich) [inline mne/io/tests/test_raw.py]: `orig_time` attribute is missing

**Comment by agramfort** [comment]
I pushed a cosmit. this looks clean but we need to finish/merge https://github.com/mne-tools/mne-python/pull/4870 first.

cc @massich

**Comment by agramfort** [comment]
I took this code in https://github.com/mne-tools/mne-python/pull/4936
which also tries to unify with the EEGLAB approach

**Comment by massich** [comment]
closed via #4936

**Pull request closed without merging**